File: Programmers/juyonLee/구명보트.cpp
//42885

#include <string>
#include <vector>
#include <algorithm>
#include <deque>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;

    sort(people.begin(), people.end());
    
    deque<int> dp(people.begin(), people.end());

    while(!dp.empty())
    {
        int n1 = dp.front();
        int n2 = dp.back();

        if(n1 + n2 > limit)
        {
            answer += 1;
            dp.pop_back();
        }
        
        else
        {
            answer += 1;
            dp.pop_back();
            if(dp.empty())
                break;
            dp.pop_front();
        }
    }
    
    return answer;
}

File: Programmers/yerim/구명보트.py
def solution(people, limit):
    answer = 0
    people.sort()
    l, r = 0, len(people)-1
    
    while l < r:
        if(people[l] + people[r] <= limit):
            answer += 1
            l += 1
            r -= 1
        else:
            answer += 1;
            r -= 1
            
    if l == r:
        answer += 1
        
    return answer